parlai/core/params.py

```diff
@@ -362,17 +376,36 @@ def add_parlai_args(self, args=None):
             help='Whether the agent should preprocess the data while building'
                  'the pytorch data')
         pytorch.add_argument(
-            '--batch-sort-cache', type=str,
-            choices=['pop', 'index', 'none'], default='none',
-            help='Whether to have batches of similarly sized episodes, and how'
-            'to build up the cache')
+            '-pybsrt', '--pytorch-teacher-batch-sort',
+            type='bool', default=False,
+            help='Whether to construct batches of similarly sized episodes'
+            'when using the PytorchDataTeacher (either via specifying `-pyt`'
+            'or `-pytd`')
+        pytorch.add_argument(
+            '--batch-sort-cache-type', type=str,
+            choices=['pop', 'index', 'none'], default='pop',
+            help='how to build up the batch cache')
         pytorch.add_argument(
             '--batch-length-range', type=int, default=5,
             help='degree of variation of size allowed in batch')
         pytorch.add_argument(
             '--shuffle', type='bool', default=False,
             help='Whether to shuffle the data')
-        self.add_parlai_data_path(parlai)
+        pytorch.add_argument(
+            '--batch-sort-field', type=str, default='text',
+            help='What field to use when determining the length of an episode')
```

Comment:
maybe set `choices=['text']`?

Reply:
fwiw: the way batch sorting is implemented in the paper require more complex operations than you have below--other options wouldn't just be e.g. "labels" but could be something like "text then labels" (currently what the regular batch sorting is doing, first sort by text then break ties by average length of the label)